Update CHANGELOG.md

Math-Bee opened the pull request:
### Summary
Updates the changelog to have more recent information
<!--
^^^ Summarize the changes done and why they were done above.

By submitting this pull request, you certify that you have read the
[CONTRIBUTING.md](https://github.com/TagStudioDev/TagStudio/blob/main/CONTRIBUTING.md).

IMPORTANT FOR FEATURES: Please verify that a feature request or some other form
of communication with maintainers was already conducted in terms of approving.

Thank you for your eagerness to contribute!
-->

### Tasks Completed

<!-- No requirements, just context for reviewers. -->

-   Platforms Tested:
    -   [ ] Windows x86
    -   [ ] Windows ARM
    -   [ ] macOS x86
    -   [ ] macOS ARM
    -   [ ] Linux x86
    -   [ ] Linux ARM
    <!-- If an unspecified platform was tested, please add it here -->
-   Tested For:
    -   [ ] Basic functionality
    -   [ ] PyInstaller executable <!-- Not necessarily required, but appreciated! -->
    <!-- If other important criteria was tested for, please add it here -->

CyanVoxel commented:
Sorry for the delay on getting around to this!
Thank you for the partial update! For reference, the changelogs have been a part of the [Releases](https://github.com/TagStudioDev/TagStudio/releases) page since the last time this file was updated. I could merge this now unless you'd be interested in further updating the file with the newer versions

Math-Bee commented:
Just added the new parts! A few notes though:
1. I copied everything to the changelog except for the "Still Coming in v9.5.0" and "New Contributors" sections, because I didn't think that they needed to go in a changelog
2.  On the 9.5.0 Pre-Release 4 version, the "Fixed" and "Changed" sections had the wrong heading level (h3 instead of h2) so I fixed them here
